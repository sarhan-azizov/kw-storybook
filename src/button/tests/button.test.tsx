import React from 'react';
import { shallow } from 'enzyme';

import { Button, BUTTON_SIZES, BUTTON_KINDS } from '../index';

describe('Button component', () => {
    const requiredProps = {
        children: <span>content</span>,
        onClick: jest.fn()
    };

    it('should render component with default props', () => {
        const wrapper = shallow(<Button {...requiredProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with kind prop', () => {
        const wrapper = shallow(<Button {...requiredProps} kind={BUTTON_KINDS.primary}/>);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with size prop', () => {
        const wrapper = shallow(<Button {...requiredProps} size={BUTTON_SIZES.medium}/>);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with type reset', () => {
        const wrapper = shallow(<Button {...requiredProps} type="button" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with active prop', () => {
        const wrapper = shallow(<Button {...requiredProps} active/>);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with disabled prop', () => {
        const wrapper = shallow(<Button {...requiredProps} disabled/>);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with className prop', () => {
        const wrapper = shallow(<Button {...requiredProps} className="className"/>);

        expect(wrapper).toMatchSnapshot();
    });

    it('shouldn\'t render component', () => {
        const wrapper = shallow(<Button {...requiredProps} rendered={false}/>);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render hidden component', () => {
        const wrapper = shallow(<Button {...requiredProps} hidden/>);

        expect(wrapper).toMatchSnapshot();
    });
});
