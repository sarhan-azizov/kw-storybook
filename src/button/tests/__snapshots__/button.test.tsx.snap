// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render component with active prop 1`] = `
<Memo()
  active={true}
  hidden={false}
  onClick={[MockFunction]}
>
  <span>
    content
  </span>
</Memo()>
`;

exports[`Button component should render component with className prop 1`] = `
<Memo()
  className="className"
  hidden={false}
  onClick={[MockFunction]}
>
  <span>
    content
  </span>
</Memo()>
`;

exports[`Button component should render component with default props 1`] = `
<Memo()
  hidden={false}
  onClick={[MockFunction]}
>
  <span>
    content
  </span>
</Memo()>
`;

exports[`Button component should render component with disabled prop 1`] = `
<Memo()
  disabled={true}
  hidden={false}
  onClick={[MockFunction]}
>
  <span>
    content
  </span>
</Memo()>
`;

exports[`Button component should render component with kind prop 1`] = `
<Memo()
  hidden={false}
  kind={0}
  onClick={[MockFunction]}
>
  <span>
    content
  </span>
</Memo()>
`;

exports[`Button component should render component with size prop 1`] = `
<Memo()
  hidden={false}
  onClick={[MockFunction]}
  size={3}
>
  <span>
    content
  </span>
</Memo()>
`;

exports[`Button component should render component with type reset 1`] = `
<Memo()
  hidden={false}
  onClick={[MockFunction]}
  type="button"
>
  <span>
    content
  </span>
</Memo()>
`;

exports[`Button component should render hidden component 1`] = `
<div
  hidden={true}
>
  <Memo()
    hidden={true}
    onClick={[MockFunction]}
  >
    <span>
      content
    </span>
  </Memo()>
</div>
`;

exports[`Button component shouldn't render component 1`] = `""`;
